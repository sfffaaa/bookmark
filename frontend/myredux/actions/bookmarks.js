import {
    CREATE_BOOKMARK_ACTION_TYPE,
    LIST_BOOKMARK_ACTION_TYPE,
    DELETE_BOOKMARK_ACTION_TYPE,
    UPGRADE_BOOKMARK_ACTION_TYPE,
} from '../constants/action-types';

const BASE_URL = 'http://localhost:3000';

export const createBookmark = url => ({
    type: 'API',
    payload: {
        bookURL: url,
        url: `${BASE_URL}/api/create`,
    },
    actionType: CREATE_BOOKMARK_ACTION_TYPE,
});

export const listBookmark = () => ({
    type: 'API',
    payload: {
        url: `${BASE_URL}/api/list`,
    },
    actionType: LIST_BOOKMARK_ACTION_TYPE,
});

export const deleteBookmark = bookmarkId => ({
    type: 'API',
    payload: {
        url: `${BASE_URL}/api/delete`,
        id: bookmarkId,
    },
    actionType: DELETE_BOOKMARK_ACTION_TYPE,
});

export const upgradeBookmark = bookmarkId => ({
    type: 'API',
    payload: {
        url: `${BASE_URL}/api/upgrade`,
        id: bookmarkId,
    },
    actionType: UPGRADE_BOOKMARK_ACTION_TYPE,
});
