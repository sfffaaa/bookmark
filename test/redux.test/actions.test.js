/* global describe it expect */
import * as actions from '../../frontend/myredux/actions/bookmarks';
import { MYURL } from './constant';

describe('Redux action test', () => {
    it('should create an action to add a bookmark', () => {
        expect(actions.createBookmark(MYURL)).toMatchSnapshot();
    });

    it('should create an action to list a bookmark', () => {
        expect(actions.listBookmark()).toMatchSnapshot();
    });
});
