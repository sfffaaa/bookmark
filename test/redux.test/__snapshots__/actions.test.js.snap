// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux action test should create an action to add a bookmark 1`] = `
Object {
  "actionType": Object {
    "ERROR": "CREATE_BOOKMARK_ERROR",
    "PENDING": "CREATE_BOOKMARK_PENDING",
    "SUCCESS": "CREATE_BOOKMARK_SUCCESS",
    "TYPE": "CREATE_BOOKMARK_TYPE",
  },
  "payload": Object {
    "bookURL": "http://NOUSEURL",
    "url": "http://localhost:3000/api/create",
  },
  "type": "API",
}
`;

exports[`Redux action test should create an action to list a bookmark 1`] = `
Object {
  "actionType": Object {
    "ERROR": "LIST_BOOKMARK_ERROR",
    "PENDING": "LIST_BOOKMARK_PENDING",
    "SUCCESS": "LIST_BOOKMARK_SUCCESS",
    "TYPE": "LIST_BOOKMARK_TYPE",
  },
  "payload": Object {
    "url": "http://localhost:3000/api/list",
  },
  "type": "API",
}
`;
